telegram/_chatfullinfo.py

```diff
@@ -490,10 +490,10 @@ def __init__(
             self.unrestrict_boost_count: Optional[int] = unrestrict_boost_count
             self.custom_emoji_sticker_set_name: Optional[str] = custom_emoji_sticker_set_name
             self.birthdate: Optional[Birthdate] = birthdate
-            self.personal_chat: Optional["Chat"] = personal_chat
-            self.business_intro: Optional["BusinessIntro"] = business_intro
-            self.business_location: Optional["BusinessLocation"] = business_location
-            self.business_opening_hours: Optional["BusinessOpeningHours"] = business_opening_hours
+            self.personal_chat: Optional[Chat] = personal_chat
```

Comment:
I'm wondering about these change. I gguess you didn't remove *all* double quotes on types - otherwise we'd probably get a bunch of import errors? So these instances where marked as superfluous by ruff? Why was this marked only after adding py3.13 and not earlier? note that the type completeness workflow runs on py3.9. updating pyright in the workflow might also be an option.

Reply:
> otherwise we'd probably get a bunch of import errors? So these instances where marked as superfluous by ruff?

yeah, these were autocorrected by ruff.

> Why was this marked only after adding py3.13 and not earlier? 

probably because we missed updating the `target_version` for python 3.12 - https://github.com/python-telegram-bot/python-telegram-bot/pull/4253/commits/585b362ad4773c3121408ac2efb9b090fc3ca0f6#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711. Now that it autodetects it, this shouldn't happen again.

> note that the type completeness workflow runs on py3.9. updating pyright in the workflow might also be an option

ok let me upgrade it and check. I can't seem to run it locally btw, don't know what I'm doing wrong: 

```
Module name: "telegram"
Package directory: ""
Module directory: ""
   error: No py.typed file found

Symbols used in public interface:

Symbols exported by "telegram": 0
  With known type: 0
  With ambiguous type: 0
  With unknown type: 0
    (Ignoring unknown types imported from other packages)

```

Reply:
> probably because we missed updating the target_version for python 3.12

dedum …

> I can't seem to run it locally btw, don't know what I'm doing wrong:

you need to install ptb locally as done in the workflow, i.e. `pip install .`

The quotes are a bit confusing and I'm looking forward to and https://peps.python.org/pep-0563/ being implemented by default (not only via `__future__`). e.g. in chatfullinfo the business* classes are imported only in the `if TYPE_CHECKING` clause 

https://github.com/python-telegram-bot/python-telegram-bot/blob/eca420ae4db56d104fa19d14a8cade2c709ca24d/telegram/_chatfullinfo.py#L34-L35

so that they are not available at definition time of the class and  `__init__` has to use the quotes

https://github.com/python-telegram-bot/python-telegram-bot/blob/eca420ae4db56d104fa19d14a8cade2c709ca24d/telegram/_chatfullinfo.py#L404-L406

But for some reason, it's okay to drop the quotes in the boday of `__init__`:

https://github.com/python-telegram-bot/python-telegram-bot/blob/eca420ae4db56d104fa19d14a8cade2c709ca24d/telegram/_chatfullinfo.py#L493-L496

Trying to insert `print(BusinessIntro)` or `print(Optional[BI])` in the `__init__` raises a `NameError` as expected. WTF. And why are the quotes removed by ruff only if we add py3.12 support? If that's a new feature in 3.12 that's not supported in 3.11-, ruff should stay away from that or the tests should fail.

So many questions 🤯 Anyways, as long as the tests succedd I'm happy :D

Reply:
Regarding type completeness it looks to me like pyright isn't happey with `JobQueue.scheduler for some reason

```
telegram.ext._jobqueue.JobQueue.scheduler
  c:\Users\hinri\PycharmProjects\python-telegram-bot\venv311\Lib\site-packages\telegram\ext\_jobqueue.py:111:14 - error: Type unknown for variable "telegram.ext._jobqueue.JobQueue.scheduler"
```

That makes the type of `JobQueue` partially unknown and also for every class that `JobQueue` is referenced in … I haven't really figured out what pyrights problem is with aps though …